search/exponential_search.ts

```diff
@@ -26,24 +33,9 @@ export const exponentialSearch = (
     i = i * 2
   }
 
-  return binarySearch(array, x, i / 2, Math.min(i, arrayLength - 1))
-}
+  const start = Math.floor(i / 2)
+  const end = Math.min(i, arrayLength - 1)
+  const result = binarySearchIterative(array, x, start, end)
 
-const binarySearch = (
-  array: number[],
-  x: number,
-  start: number,
-  end: number
-): number | null => {
-  while (start <= end) {
-    const mid = Math.floor((start + end) / 2)
-    if (array[mid] === x) {
-      return mid
-    } else if (array[mid] < x) {
-      start = mid + 1
-    } else {
-      end = mid - 1
-    }
-  }
-  return null
+  return result !== -1 ? result : null
```

Comment:
Feel free to refactor the binary search to return `null` if the element isn't present (this will require you to update the tests)